addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -3880,6 +3880,19 @@ export class OdooEditor extends EventTarget {
             if (!focusNode) {
                 return;
             }
+            // We are adding a zws before the code element and set selection next to it.
+            // Because on delete forward the zws before the code element will be removed and
+            // when we try to hit left arrow cursor won't get out of the code block.
```

Comment:
Nice but I wouldn't mention the delete part since it's not directly relevant (this could happen in different ways): we just want to be sure there's a zws when there isn't one yet.

```suggestion
            // If the selection is at the beginning of a code element at the
            // start of its parent, make sure there's a zws before it, where the
            // selection can then be set.
```